src/Microsoft.DotNet.VersionTools/Automation/VersionsRepoUpdater.cs

```diff
@@ -68,25 +71,71 @@ public VersionsRepoUpdater(GitHubAuth gitHubAuth, GitHubProject project)
                 .FirstOrDefault(prerelease => !string.IsNullOrEmpty(prerelease))
                 ?? "";
 
-            if (updateLatestPackageList)
+            Dictionary<string, string> packageDictionary = packages.ToDictionary(
+                    t => t.Id,
+                    t => t.Version);
+
+            using (GitHubClient client = new GitHubClient(_gitHubAuth))
             {
-                string packageInfoFileContent = string.Join(
-                    Environment.NewLine,
-                    packages
-                        .OrderBy(t => t.Id)
-                        .Select(t => $"{t.Id} {t.Version}"));
+                // Master commit to use as new commit's parent.
+                string masterRef = "heads/master";
+                GitReference currentMaster = await client.GetReferenceAsync(_project, masterRef);
+                string masterSha = currentMaster.Object.Sha;
 
-                string packageInfoFilePath = $"{versionsRepoPath}/Latest_Packages.txt";
-                string message = $"Updating Latest_Packages.txt at {versionsRepoPath} for {prereleaseVersion}";
+                List<GitObject> objects = new List<GitObject>();
 
-                await UpdateGitHubFileAsync(packageInfoFilePath, packageInfoFileContent, message);
-            }
+                if (updateLastBuildPackageList)
+                {
+                    objects.Add(new GitObject
+                    {
+                        Path = $"{versionsRepoPath}/Last_Build_Packages.txt",
+                        Type = GitObject.TypeBlob,
+                        Mode = GitObject.ModeFile,
+                        Content = CreatePackageListFile(packageDictionary)
+                    });
+                }
 
-            if (updateLatestVersion)
-            {
-                string latestFilePath = $"{versionsRepoPath}/Latest.txt";
+                if (updateLatestPackageList)
+                {
+                    string path = $"{versionsRepoPath}/Latest_Packages.txt";
+
+                    var allPackages = new Dictionary<string, string>(packageDictionary);
+
+                    if (updateLastBuildPackageList)
+                    {
+                        Dictionary<string, string> existingPackages = await GetPackagesAsync(client, path);
```

Comment:
Sorry I miss-read this as I now realize you are getting the Lastest_Packages.txt file instead. I would consider changing the name of path to latestPackagesPath.